### ISSUE TITLE
Fix accessibility of methods violations caused by bump of spotbugs-maven-plugin from 4.9.8.5 to 4.10.2.0

### ISSUE THREAD

**Issue by SteLeo1602** [checkstyle/checkstyle#20482]
Detected in https://github.com/checkstyle/checkstyle/pull/20310

Instance error messages from spotbugs-and-pmd CI:

```
[INFO] --- spotbugs:4.10.2.0:check (default-cli) @ checkstyle ---
[INFO] BugInstance size is 15
[INFO] Error size is 0
[INFO] Total bugs: 15
[ERROR] Low: Accessibility of the overridden or hidden method com.puppycrawl.tools.checkstyle.TreeWalker.finishLocalSetup() was increased from protected to public. [com.puppycrawl.tools.checkstyle.TreeWalker] At TreeWalker.java:[lines 132-136] IAOM_DO_NOT_INCREASE_METHOD_ACCESSIBILITY
[ERROR] Low: Accessibility of the overridden or hidden method com.puppycrawl.tools.checkstyle.TreeWalker.setupChild(Configuration) was increased from protected to public. [com.puppycrawl.tools.checkstyle.TreeWalker] At TreeWalker.java:[lines 144-168] IAOM_DO_NOT_INCREASE_METHOD_ACCESSIBILITY
[ERROR] Low: Accessibility of the overridden or hidden method com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$AnonymousClassFrame.getFrameName() was increased from package private to protected. [com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$AnonymousClassFrame] At RequireThisCheck.java:[line 1556] IAOM_DO_NOT_INCREASE_METHOD_ACCESSIBILITY
[ERROR] Low: Accessibility of the overridden or hidden method com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$BlockFrame.getType() was increased from package private to protected. [com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$BlockFrame] At RequireThisCheck.java:[line 1578] IAOM_DO_NOT_INCREASE_METHOD_ACCESSIBILITY
[ERROR] Low: Accessibility of the overridden or hidden method com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$CatchFrame.getIfContains(DetailAST, boolean) was increased from package private to protected. [com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$CatchFrame] At RequireThisCheck.java:[lines 1607-1618] IAOM_DO_NOT_INCREASE_METHOD_ACCESSIBILITY
[ERROR] Low: Accessibility of the overridden or hidden method com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$CatchFrame.getType() was increased from package private to public. [com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$CatchFrame] At RequireThisCheck.java:[line 1600] IAOM_DO_NOT_INCREASE_METHOD_ACCESSIBILITY
[ERROR] Low: Accessibility of the overridden or hidden method com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$ClassFrame.containsFieldOrVariable(DetailAST) was increased from package private to protected. [com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$ClassFrame] At RequireThisCheck.java:[line 1458] IAOM_DO_NOT_INCREASE_METHOD_ACCESSIBILITY
[ERROR] Low: Accessibility of the overridden or hidden method com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$ClassFrame.getIfContains(DetailAST, boolean) was increased from package private to protected. [com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$ClassFrame] At RequireThisCheck.java:[lines 1470-1479] IAOM_DO_NOT_INCREASE_METHOD_ACCESSIBILITY
[ERROR] Low: Accessibility of the overridden or hidden method com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$ClassFrame.getType() was increased from package private to protected. [com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$ClassFrame] At RequireThisCheck.java:[line 1360] IAOM_DO_NOT_INCREASE_METHOD_ACCESSIBILITY
[ERROR] Low: Accessibility of the overridden or hidden method com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$ClassFrame.isProperDefinition(DetailAST, DetailAST) was increased from package private to protected. [com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$ClassFrame] At RequireThisCheck.java:[lines 1464-1465] IAOM_DO_NOT_INCREASE_METHOD_ACCESSIBILITY
[ERROR] Low: Accessibility of the overridden or hidden method com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$ConstructorFrame.getType() was increased from package private to protected. [com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$ConstructorFrame] At RequireThisCheck.java:[line 1325] IAOM_DO_NOT_INCREASE_METHOD_ACCESSIBILITY
[ERROR] Low: Accessibility of the overridden or hidden method com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$ForFrame.getType() was increased from package private to public. [com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$ForFrame] At RequireThisCheck.java:[line 1640] IAOM_DO_NOT_INCREASE_METHOD_ACCESSIBILITY
[ERROR] Low: Accessibility of the overridden or hidden method com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$MethodFrame.getType() was increased from package private to protected. [com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$MethodFrame] At RequireThisCheck.java:[line 1303] IAOM_DO_NOT_INCREASE_METHOD_ACCESSIBILITY
[ERROR] Low: Accessibility of the overridden or hidden method com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$TryWithResourcesFrame.getType() was increased from package private to public. [com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$TryWithResourcesFrame] At RequireThisCheck.java:[line 1663] IAOM_DO_NOT_INCREASE_METHOD_ACCESSIBILITY
[ERROR] Low: Accessibility of the overridden or hidden method com.puppycrawl.tools.checkstyle.site.XdocsTemplateSinkFactory.createSink(Writer, String) was increased from protected to public. [com.puppycrawl.tools.checkstyle.site.XdocsTemplateSinkFactory] At XdocsTemplateSinkFactory.java:[line 46] IAOM_DO_NOT_INCREASE_METHOD_ACCESSIBILITY
[INFO] 

```

wrapped errors:
`[ERROR] Low: Accessibility of the overridden or hidden method com.puppycrawl.tools.checkstyle.TreeWalker.finishLocalSetup() was increased from protected to public. [com.puppycrawl.tools.checkstyle.TreeWalker] At TreeWalker.java:[lines 132-136] IAOM_DO_NOT_INCREASE_METHOD_ACCESSIBILITY`
`[ERROR] Low: Accessibility of the overridden or hidden method com.puppycrawl.tools.checkstyle.TreeWalker.setupChild(Configuration) was increased from protected to public. [com.puppycrawl.tools.checkstyle.TreeWalker] At TreeWalker.java:[lines 144-168] IAOM_DO_NOT_INCREASE_METHOD_ACCESSIBILITY`
`[ERROR] Low: Accessibility of the overridden or hidden method com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$AnonymousClassFrame.getFrameName() was increased from package private to protected. [com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck$AnonymousClassFrame] At RequireThisCheck.java:[line 1556] IAOM_DO_NOT_INCREASE_METHOD_ACCESSIBILITY`
`[ERROR] Low: Accessibility of the overridden or hidden method com.puppycrawl.tools.checkstyle.site.XdocsTemplateSinkFactory.createSink(Writer, String) was increased from protected to public. [com.puppycrawl.tools.checkstyle.site.XdocsTemplateSinkFactory] At XdocsTemplateSinkFactory.java:[line 46] IAOM_DO_NOT_INCREASE_METHOD_ACCESSIBILITY`

**Comment by Carbon14-48** [comment]
on it

**Comment by romani** [comment]
PR for suppression, not a fix for spotbug violations 
Is at checkstyle/checkstyle#20486

**Comment by romani** [comment]
regression happened and detcted by CI
plugins are using Treewalker a lot, so we have to keep methods public, to let them initialize Checkstyle execution.
```
[INFO] Reactor Summary for methods-distance-parent 1.0-SNAPSHOT: 01:51
[INFO] 01:51
[INFO] methods-distance-parent ............................ SUCCESS [  4.477 s] 01:51
[INFO] methods-distance ................................... FAILURE [  2.757 s] 01:51
[INFO] methods-distance-web ............................... SKIPPED 01:51
[INFO] ------------------------------------------------------------------------ 01:51
[INFO] BUILD FAILURE 01:51
[INFO] ------------------------------------------------------------------------ 01:51
[INFO] Total time:  7.290 s 01:51
[INFO] Finished at: 2026-07-17T02:07:54Z 01:51
[INFO] ------------------------------------------------------------------------ 01:51
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.3:compile (default-compile) on project methods-distance: Compilation failure: Compilation failure: 01:51
[ERROR] /home/semaphore/checkstyle/.ci-temp/methods-distance/methods-distance/src/main/java/com/github/sevntu/checkstyle/ReportCli.java:[63,11] finishLocalSetup() has protected access in com.puppycrawl.tools.checkstyle.TreeWalker 01:51
[ERROR] /home/semaphore/checkstyle/.ci-temp/methods-distance/methods-distance/src/main/java/com/github/sevntu/checkstyle/ReportCli.java:[64,11] setupChild(com.puppycrawl.tools.checkstyle.api.Configuration) has protected access in com.puppycrawl.tools.checkstyle.TreeWalker 01:51
[ERROR] /home/semaphore/checkstyle/.ci-temp/methods-distance/methods-distance/src/main/java/com/github/sevntu/checkstyle/common/MethodCallDependencyCheckInvoker.java:[54,11] finishLocalSetup() has protected access in com.puppycrawl.tools.checkstyle.TreeWalker 01:51
[ERROR] /home/semaphore/checkstyle/.ci-temp/methods-distance/methods-distance/src/main/java/com/github/sevntu/checkstyle/common/MethodCallDependencyCheckInvoker.java:[55,11] setupChild(com.puppycrawl.tools.checkstyle.api.Configuration) has protected access in com.puppycrawl.tools.checkstyle.TreeWalker 01:51
[ERROR] -> [Help 1] 01:51
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.3:compile (default-compile) on project methods-distance: Compilation failure
```
